pkg/reconciler/v1alpha1/serverlessservice/serverlessservice.go

```diff
@@ -0,0 +1,342 @@
+		FilterFunc: controller.Filter(netv1alpha1.SchemeGroupVersion.WithKind("ServerlessService")),
+		Handler:    rbase.Handler(impl.EnqueueControllerOf),
+	})
+
+	// Watch activator-service endpoints.
+	grCb := func(obj interface{}) {
+		// Since changes in the Activar Service endpoints affect all the SKS objects,
+		// do a global resync.
+		c.Logger.Info("Doing a global resync due to activator endpoint changes")
+		impl.GlobalResync(sksInformer.Informer())
+	}
+	endpointsInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		// Accept only ActivatorService K8s service objects.
+		FilterFunc: func(obj interface{}) bool {
+			if object, ok := obj.(metav1.Object); ok {
+				return object.GetNamespace() == activatorServiceNS && object.GetName() == activatorService
+			}
+			return false
+		},
+		Handler: rbase.Handler(grCb),
+	})
+
+	return impl
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*reconciler)(nil)
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Revision resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	logger := logging.FromContext(ctx)
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		logger.Errorf("Invalid resource key: %s", key)
+		return nil
+	}
+
+	logger.Debugf("Reconciling SKS resource: %s", key)
+	// Get the current SKS resource.
+	original, err := r.sksLister.ServerlessServices(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		// The resource may no longer exist, in which case we stop processing.
+		logger.Errorf("SKS resource %q in work queue no longer exists", key)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// Don't modify the informers copy.
+	sks := original.DeepCopy()
+	err = r.reconcile(ctx, sks)
+	if err != nil {
+		r.Recorder.Eventf(sks, corev1.EventTypeWarning, "UpdateFailed", "InternalError: %v", err.Error())
+		return err
+	}
+	if !equality.Semantic.DeepEqual(sks.Status, original.Status) {
+		// Only update status if it changed.
+		if _, err = r.updateStatus(sks); err != nil {
+			r.Recorder.Eventf(sks, corev1.EventTypeWarning, "UpdateFailed", "Failed to update status: %v", err)
+			return err
+		}
+		r.Recorder.Eventf(sks, corev1.EventTypeNormal, "Updated", "Successfully updated ServerlessService %q", key)
+	}
+	return nil
+}
+
+func (r *reconciler) reconcile(ctx context.Context, sks *netv1alpha1.ServerlessService) error {
+	logger := logging.FromContext(ctx)
+	// Don't reconcile if we're being deleted.
+	if sks.GetDeletionTimestamp() != nil {
+		return nil
+	}
+
+	sks.SetDefaults(ctx)
+	sks.Status.InitializeConditions()
+
+	for i, fn := range []func(context.Context, *netv1alpha1.ServerlessService) error{
```

Comment:
It's make me feel a better with a `// TODO(#1234): Probe the public service to determine that the activator is now proxying requests` wherever you plan to do the positive hand-off.

Reply:
Added the comment,.